### PR TITLE
feat: add `reth config` cmd to dump default reth.toml

### LINKED ISSUE #1897: Reth config which dumps the default reth.toml to stdout
### Describe the feature

Title pretty much captures it.

### Additional context

Based on @gakonst 's [request](https://t.me/paradigm_reth/2317) in Telegram

### PR DESCRIPTION
**Pull request by georgewhewell**
resolves #1897 

hi, first contrib to this repo, just picked a simple issue

- didn't add test as functionality is trivial
- added `toml` crate- config io is handled by `confy` but it doesn't expose any functionality to write to string, only path. could also have written to tempfile and read back, seemed cleaner like this though

### CODE REVIEW

File: bin/reth/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//! CLI command to write default config to stdout
+use clap::Parser;
+use reth_staged_sync::Config;
+
+/// `reth config` command
+#[derive(Debug, Parser)]
+pub struct Command;
```
Comment (mattsse) [inline bin/reth/src/config.rs:7]: we need a path argument, same as:

https://github.com/paradigmxyz/reth/blob/087d0a63170b944dc70435c691ec54069ccfeea2/bin/reth/src/node/mod.rs#L81-L83

File: bin/reth/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+impl Command {
+    /// Execute `config` command - write default config to stdout
+    pub async fn execute(&self) -> eyre::Result<()> {
+        let config = Config::default();
```
Comment (mattsse) [inline bin/reth/src/config.rs:12]: need to load the active config here, same as:

https://github.com/paradigmxyz/reth/blob/087d0a63170b944dc70435c691ec54069ccfeea2/bin/reth/src/node/mod.rs#L404-L404

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks!

this will print the default config, by adding the config arg (same as node cmd) we can print the active or a specific config

**Comment by georgewhewell** [comment]
thx for review @mattsse 
split into `default`  and `active` (optionally with `--config`) subcommands- is this what you meant?

File: bin/reth/src/config.rs [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
+pub enum Subcommands {
+    /// Show the default config
+    Default,
+    /// Show the active config
+    Show {
+        /// The path to the configuration file to use.
+        #[arg(long, value_name = "FILE", verbatim_doc_comment, default_value_t)]
+        config: PlatformPath<ConfigPath>,
+    },
+}
```
Comment (mattsse) [inline bin/reth/src/config.rs:26]: hmm,
I'm not convinced the subcommand is the best UX here.

looks like we have three cases:

```
reth config # print current active config
reth config <file>
reth config --default
```

so I'd like to get rid of the subcommand here and use mutually exclusive arguments, which would require:

1. `config: Option<PlatformPath>`
2. `default: bool` `conflicts_with = "config"`

**Review by mattsse — CHANGES_REQUESTED** [review]
`execute` lgtm,

suggestion re subcommand

File: bin/reth/src/config.rs (hunk 3 above)
Comment (georgewhewell) [reply bin/reth/src/config.rs]: cool, done :+1:

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**